File: sdk/src/main/java/com/cxense/cxensesdk/model/ContextParameter.java
package com.cxense.cxensesdk.model;

import androidx.annotation.NonNull;

/**
 * Context parameter that replace the placeholders are passed from the widget data request.
 *
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2017-06-05).
 */

@SuppressWarnings({"UnusedDeclaration", "WeakerAccess"}) // Public API.
public final class ContextParameter {
    /**
     * Parameter key
     */
    @NonNull
    public String key = "";

    /**
     * Parameter value
     */
    @NonNull
    public String value = "";

    private ContextParameter() {
    }

    public ContextParameter(@NonNull String key, @NonNull String value) {
        this.key = key;
        this.value = value;
    }
}

File: sdk/src/main/java/com/cxense/cxensesdk/model/ContextParameter.kt
package com.cxense.cxensesdk.model

import com.squareup.moshi.JsonClass

/**
 * Context parameter that replace the placeholders are passed from the widget data request.
 * @param key Defines the key we use for the replacement
 * @param value Defines the value associated with key or the default value
 */
@Suppress("unused", "MemberVisibilityCanBePrivate") // Public API.
@JsonClass(generateAdapter = true)
class ContextParameter(
    val key: String,
    val value: String
)
